crates/optimism/chainspec/src/base.rs

```diff
@@ -1,18 +1,17 @@
 //! Chain specification for the Base Mainnet network.
 
-use alloc::sync::Arc;
+use alloc::{sync::Arc, vec};
```

Comment:
how come you have to import `vec` module?

Reply:
I want a macro here: https://doc.rust-lang.org/alloc/macro.vec.html

Reply:
right, for l26-29
```rust
vec![
    (EthereumHardfork::London.boxed(), BaseFeeParams::optimism()),
    (OptimismHardfork::Canyon.boxed(), BaseFeeParams::optimism_canyon()),
]
```
tbh that would have been better off as it's own pr

Reply:
Yes, agreed. But `once_cell` -> std on full migration needs that crate(chainspec) to be no_std. That’s why I put no_std with `alloc` of here and shift `once_cell` to be no_std in workspace Cargo.toml by disabling default features and enabled `critical-section` that used for no_std blocking

Reply:
that `vec` pr could have come before this pr :)

Reply:
@emhane do you think it's good just add `vec` on PR everywhere on `optimism/chainspec` crate, or also make it no_std via
```rust
#![cfg_attr(not(feature = "std"), no_std)]
``` 
on this PR? Also with `std::fmt` -> `core::fmt` and `std::sync::Arc` -> `alloc::sync::Arc` and so on.

Reply:
> @emhane do you think it's good just add `vec` on PR everywhere on `optimism/chainspec` crate, or also make it no_std via
> 
> ```rust
> #![cfg_attr(not(feature = "std"), no_std)]
> ```
> 
> on this PR? Also with `std::fmt` -> `core::fmt`.

Also I propose `std` feature on `optimism/chainspec` to be like
```toml
[features]
default = ["std"]
std = [
    "alloy-chains/std",
    "alloy-genesis/std",
    "alloy-primitives/std",
    "op-alloy-rpc-types/std",
    "reth-chainspec/std",
    "reth-ethereum-forks/std",
    "reth-primitives-traits/std",
    "reth-optimism-forks/std",
]
```

instead of:

```toml
[features]
default = ["std"]
std = []
```

in that case

Reply:
the pr is fine now as is, but for future ref, smaller prs will more likely get merged because they will be reviewed quicker. so a good rule of thumb is, **if** you can get a subset of the implementation compiling, then open a pr with that subset first. in the pr description just add "Ref <issue>", instead of "Closes <issue>".

Reply:
thank you! I'll aim for smaller PRs in the future and will use "Ref" for subsets that are compiling. Appreciate the feedback